src/main/resources/database/postgresql/1/patch_24.sql

```diff
@@ -0,0 +1,178 @@
+-- feature/static-metrics
+-- Create a new table containing the reputation of each day grouped by day
+CREATE TABLE repbot_schema.metrics_reputation
+(
+    day   DATE   NOT NULL,
+    cause TEXT   NOT NULL,
+    count BIGINT NOT NULL
+);
+
+CREATE INDEX metrics_reputation_day_index
+    ON repbot_schema.metrics_reputation (day);
+
+ALTER TABLE repbot_schema.metrics_reputation
+    ADD CONSTRAINT metrics_reputation_pk
+        PRIMARY KEY (day, cause);
+
+-- prepopulate the table with data from the not current day. After this the metric service takes over.
+INSERT INTO repbot_schema.metrics_reputation
+SELECT received::DATE AS day,
+       cause,
+       COUNT(1)       AS count
+FROM repbot_schema.reputation_log
+WHERE received::DATE != NOW()::DATE
+GROUP BY day, cause
+ORDER BY day DESC;
+
+-- We now need to recreate all views which were based on the reputation log itself
+CREATE OR REPLACE VIEW repbot_schema.metrics_reputation_type_week AS
+SELECT DATE_TRUNC('week', day)::DATE AS week,
+       cause,
+       SUM(count)::BIGINT            AS count
+FROM repbot_schema.metrics_reputation
+GROUP BY week, cause
+ORDER BY week DESC, cause;
+
+CREATE OR REPLACE VIEW repbot_schema.metrics_reputation_type_month AS
+SELECT DATE_TRUNC('month', day)::DATE AS month,
+       cause,
+       SUM(count)::BIGINT             AS count
+FROM repbot_schema.metrics_reputation
+GROUP BY month, cause
+ORDER BY month DESC;
+
+CREATE OR REPLACE VIEW repbot_schema.metrics_reputation_dow_week AS
+SELECT DATE_TRUNC('week', day)::DATE AS week,
+       EXTRACT(ISODOW FROM day)      AS dow,
+       SUM(count)::BIGINT            AS count
+FROM repbot_schema.metrics_reputation
+GROUP BY week, dow
+ORDER BY week DESC, dow ASC;
+
+CREATE OR REPLACE VIEW repbot_schema.metrics_reputation_week AS
```

Comment:
No its not the same since the type view does also group by type. This view is basically avoiding an unrequired aggregate. If we would use the type view as a base here we would have an aggregate which we dont even use. Thats why there are two views for this.

Reply:
Not necessarily, but we need to maintain two views anyway, so we can just go with the "optimised" one. 

Query plan wise there is no large difference at the moment, beside the memory consumption
Without cause
```
Sort  (cost=171.96..174.62 rows=1067 width=12) (actual time=1.402..1.408 rows=196 loops=1)
  Sort Key: ((date_trunc('week'::text, (day)::timestamp with time zone))::date) DESC
  Sort Method: quicksort  Memory: 34kB
  ->  HashAggregate  (cost=94.28..118.29 rows=1067 width=12) (actual time=1.334..1.376 rows=196 loops=1)
        Group Key: (date_trunc('week'::text, (day)::timestamp with time zone))::date
        Batches: 1  Memory Usage: 97kB
        ->  Seq Scan on metrics_reputation  (cost=0.00..78.00 rows=3257 width=12) (actual time=0.013..0.896 rows=3257 loops=1)
Planning Time: 0.069 ms
Execution Time: 1.443 ms
```

With cause
```
Sort  (cost=180.10..182.77 rows=1067 width=19) (actual time=1.884..1.910 rows=602 loops=1)
  Sort Key: ((date_trunc('week'::text, (day)::timestamp with time zone))::date) DESC, cause
  Sort Method: quicksort  Memory: 72kB
  ->  HashAggregate  (cost=102.42..126.43 rows=1067 width=19) (actual time=1.607..1.737 rows=602 loops=1)
        Group Key: (date_trunc('week'::text, (day)::timestamp with time zone))::date, cause
        Batches: 1  Memory Usage: 193kB
        ->  Seq Scan on metrics_reputation  (cost=0.00..78.00 rows=3257 width=19) (actual time=0.011..0.960 rows=3257 loops=1)
Planning Time: 0.053 ms
Execution Time: 1.953 ms
```

Using the view with cause and do a second aggretation
```
HashAggregate  (cost=198.77..201.27 rows=200 width=36) (actual time=1.909..1.940 rows=196 loops=1)
  Group Key: ((date_trunc('week'::text, (metrics_reputation.day)::timestamp with time zone))::date)
  Batches: 1  Memory Usage: 64kB
  ->  Sort  (cost=180.10..182.77 rows=1067 width=19) (actual time=1.783..1.801 rows=602 loops=1)
        Sort Key: ((date_trunc('week'::text, (metrics_reputation.day)::timestamp with time zone))::date) DESC, metrics_reputation.cause
        Sort Method: quicksort  Memory: 72kB
        ->  HashAggregate  (cost=102.42..126.43 rows=1067 width=19) (actual time=1.503..1.631 rows=602 loops=1)
              Group Key: (date_trunc('week'::text, (metrics_reputation.day)::timestamp with time zone))::date, metrics_reputation.cause
              Batches: 1  Memory Usage: 193kB
              ->  Seq Scan on metrics_reputation  (cost=0.00..78.00 rows=3257 width=19) (actual time=0.011..0.899 rows=3257 loops=1)
Planning Time: 0.064 ms
Execution Time: 1.973 ms
```